mev_inspect/aave_liquidations.py

```diff
@@ -0,0 +1,128 @@
+
+        if isinstance(trace, DecodedCallTrace):
+
+            # Check for liquidation and register trace and unique liquidator
+            if (
+                trace.classification == Classification.liquidate
+                and trace.transaction_hash not in tx
+            ):
+
+                liquidations.append(trace)
+                contract_addresses.append(trace.from_address)
+                tx.append(trace.transaction_hash)
+
+                # Found liquidation, now parse inputs for data
+                for input in trace.inputs:
+
+                    if input == "_purchaseAmount":
+                        liquidation_amount = trace.inputs[input]
+                    elif input == "_collateral":
+                        collateral_address = trace.inputs[input]
+                        # This will be the address of the sent token
+                    elif input == "_reserve":
+                        reserve = trace.inputs[input]
+                        # This will be the address of the received token
+                    elif input == "_user":
+                        liquidated_user = trace.inputs[input]
+                # Register liquidation
+                result.append(
+                    Liquidation(
+                        liquidated_user=liquidated_user,
+                        collateral_address=collateral_address,
+                        collateral_amount=liquidation_amount,
+                        collateral_source="",
+                        reserve=reserve,
+                    )
+                )
+
+            elif is_transfer_from_liquidator(trace, unique_transfer_hashes):
+
+                # Add the transfer
+                liquidator = next(
+                    (
+                        contract_addresses[i]
+                        for i in range(len(contract_addresses))
+                        if trace.inputs["sender"] == contract_addresses[i]
+                    ),
+                    "",
+                )
+
+                transfers_from.append(
+                    ["from", liquidator, trace.transaction_hash, trace.inputs["amount"]]
+                )
+                unique_transfer_hashes.append(trace.transaction_hash)
+
+            elif is_transfer_to_liquidator(trace):
+
+                # Add the transfer
+                liquidator = next(
+                    (
+                        contract_addresses[i]
+                        for i in range(len(contract_addresses))
+                        if trace.inputs["recipient"] == contract_addresses[i]
+                    ),
+                    "",
+                )
+                transfers_to.append(
+                    ["to", liquidator, trace.transaction_hash, trace.inputs["amount"]]
+                )
+
+    return [result, transfers_to, transfers_from]
+
+
+def is_transfer_from_liquidator(
+    trace: ClassifiedTrace,
+    unique_transfer_hashes: List[str],
+) -> bool:
+    """Check if transfer is from liquidator"""
```

Comment:
no need for this - it's the same as the function name

Reply:
PEP 257:
"All modules should normally have docstrings, and all functions and classes exported by a module should also have docstrings."